lib/thesis/stores/cache_store.ex

```diff
@@ -0,0 +1,63 @@
+defmodule Thesis.CacheStore do
+  @moduledoc """
+  Thesis.CacheStore is a proxy store that acts in conjunction with a proper
+  store like Thesis.EctoStore to provide caching and faster page load times.
+  It can be enabled by setting `Thesis.Config.enable_cache` is set to `true`.
+  """
+
+  @behaviour Thesis.Store
+
+  import Thesis.Config
+  alias Thesis.{Page, PageContent}
+
+  def page(slug) do
+    ConCache.get_or_store(cache_name(), slug, fn() ->
+      store().page(slug)
+    end)
+  end
+
+  def page_contents(slug) when is_binary(slug) do
+    page_contents(page(slug))
+  end
+
+  def page_contents(nil) do
+    cache_get(:global, fn() ->
+      store().page_contents(nil)
+    end)
+  end
+
+  def page_contents(page = %Page{id: page_id}) do
```

Comment:
Let's decouple this from `Thesis.EctoStore` and use a map instead of `%Page` here.

```elixir
def page_contents(page = %{id: page_id}) do
```